src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,366 @@
+[Arxiv: 10.48550/arXiv.2403.05144](https://doi.org/10.48550/arXiv.2403.05144)
+"""
+mutable struct PairedExplicitRK3 <: AbstractPairedExplicitRKSingle
+    const num_stages::Int # S
+
+    a_matrix::Matrix{Float64}
+    c::Vector{Float64}
+end # struct PairedExplicitRK3
+
+# Constructor for previously computed A Coeffs
+function PairedExplicitRK3(num_stages, base_path_a_coeffs::AbstractString;
+                           cS2 = 1.0f0)
+    a_matrix, c = compute_PairedExplicitRK3_butcher_tableau(num_stages,
+                                                            base_path_a_coeffs;
+                                                            cS2)
+
+    return PairedExplicitRK3(num_stages, a_matrix, c)
+end
+
+# Constructor that computes Butcher matrix A coefficients from a semidiscretization
+function PairedExplicitRK3(num_stages, tspan, semi::AbstractSemidiscretization;
+                           verbose = false, cS2 = 1.0f0)
+    eig_vals = eigvals(jacobian_ad_forward(semi))
+
+    return PairedExplicitRK3(num_stages, tspan, eig_vals; verbose, cS2)
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a list of eigenvalues
+function PairedExplicitRK3(num_stages, tspan, eig_vals::Vector{ComplexF64};
+                           verbose = false, cS2 = 1.0f0)
+    a_matrix, c = compute_PairedExplicitRK3_butcher_tableau(num_stages,
+                                                            tspan,
+                                                            eig_vals;
+                                                            verbose, cS2)
+    return PairedExplicitRK3(num_stages, a_matrix, c)
+end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L77
+# This implements the interface components described at
+# https://diffeq.sciml.ai/v6.8/basics/integrator/#Handing-Integrators-1
+# which are used in Trixi.
+mutable struct PairedExplicitRK3Integrator{RealT <: Real, uType, Params, Sol, F, Alg,
+                                           PairedExplicitRKOptions} <:
+               AbstractPairedExplicitRKSingleIntegrator
+    u::uType
+    du::uType
+    u_tmp::uType
+    t::RealT
+    tdir::RealT
+    dt::RealT # current time step
+    dtcache::RealT # manually set time step
+    iter::Int # current number of time steps (iteration)
+    p::Params # will be the semidiscretization from Trixi
+    sol::Sol # faked
+    f::F
+    alg::Alg # This is our own class written above; Abbreviation for ALGorithm
+    opts::PairedExplicitRKOptions
+    finalstep::Bool # added for convenience
+    dtchangeable::Bool
+    force_stepfail::Bool
+    # PairedExplicitRK stages:
+    k1::uType
+    k_higher::uType
+end
+
+function init(ode::ODEProblem, alg::PairedExplicitRK3;
+              dt, callback = nothing, kwargs...)
```

Comment:
```suggestion
              dt, callback::Union{CallbackSet, Nothing} = nothing, kwargs...)
```
See below.

Reply:
In https://github.com/trixi-framework/Trixi.jl/pull/2026 we have changed that for the other integrators. So I think we can already adapt to this in this PR. Could you apply the changes done in the other PR also here, please?